runtime/src/main/java/org/corfudb/recovery/FastObjectLoader.java

```diff
@@ -632,8 +643,12 @@ private void findCheckPointsInLogAddress(long address, ILogData logData) {
      *
      */
     private void resurrectCheckpoints() {
-        streamsMetaData.entrySet().parallelStream()
-                .forEach(entry -> {
+        ExecutorService executorService = Executors.newFixedThreadPool(numOfWorkers, new ThreadFactoryBuilder()
```

Comment:
Maybe add a bit comment to the method doc highlighting the behavior that it throws FastObjectException if submit fails.

Reply:
Why would submit fail? Tasks are not submitted after shutdown is called